chromadb/api/models/CollectionCommon.py

```diff
@@ -261,6 +261,22 @@ def _validate_and_prepare_embedding_set(
         Optional[Documents],
         Optional[URIs],
     ]:
+        if ids is None:
+            if embeddings:
+                set_size = len(embeddings)
```

Comment:
will this work correctly if embeddings is actually just one embedding? Since this is `OneOrMany` and `len` on a single embedding will give you the length of the vector.

Reply:
you're right, good argument for moving this to property tests

Reply:
I think this will consume unnecessary entropy in the property tests, and should instead live on a better api test which looks for these kinds of exceptions. See #2586

Reply:
I made the new test a property test, exercises this path 379207de

Reply:
To be clear, I don't think this should be a property test. This is testing for a specific, known case that should throw an error. We only need a one-off test that should check that that error is thrown when we enter that case. Property testing is I think too heavyweight here.

Reply:
the property tests uncovered a few other edge cases that I didn't think about, so I would be in favor of keeping them, they only take a few seconds to run